### PR TITLE
When registering Ident use, consider the original span for synthetics

### LINKED ISSUE #19970: No Semanticdb generated for lifted args
## Compiler version
3.4.0
## Minimized code
Semanticdb does not include occurences in lifted args.
It can affect also other scenarios, but this one is a nice minimisation.
```Scala
object T:
  def enrichedCompilerCompletions() =
    val compilerCompletions: Seq[String] = ???

    compilerCompletions
      .toList
      .filterInteresting()

  extension (l: List[String])
    def filterInteresting(qualType: String = ""): List[String] = ???
```
## Output
```scala
[[syntax trees at end of                     typer]] 
package <empty>#106 {
  final lazy module val T: T#123 = new T#123#124#125()#126#127
  final module class T() extends Object#181#186#187()#188 {
    this: T#128.type#129 =>
    def enrichedCompilerCompletions(): List[String]#386 =
      {
        val compilerCompletions: Seq#37[String#42]#44 = ???#48#51
        {
          val l$1: List[String]#350 = compilerCompletions#52.toList#54#352
          T.filterInteresting#344(l$1#353)#355(
            this#357.filterInteresting$default$2#358(l$1#353)#359)#367
        }#368
      }#370
    #388
    extension (l: List#64[String#69]#71#76) def filterInteresting(
      qualType: String#79#150): List#87[String#92]#94 = ???#98#152
    def filterInteresting$default$2(l: List#64[String#69]#71#360):
      String @uncheckedVariance#365 = ""#83#389
  }#393
}#394
```

With semanticdb:
```scala
Test.scala
----------

Summary:
Schema => SemanticDB v4
Uri => Test.scala
Text => empty
Language => Scala
Symbols => 9 entries
Occurrences => 17 entries

Symbols:
_empty_/T. => final object T extends Object { self: T.type => +4 decls }
_empty_/T.enrichedCompilerCompletions(). => method enrichedCompilerCompletions(): List[String]
_empty_/T.filterInteresting$default$2(). => method filterInteresting$default$2(l: List[String]): String @uncheckedVariance
_empty_/T.filterInteresting$default$2().(l) => param l: List[String]
_empty_/T.filterInteresting(). => method filterInteresting(l: List[String])(qualType: String): List[String]
_empty_/T.filterInteresting().(l) => param l: List[String]
_empty_/T.filterInteresting().(qualType) => param qualType: String
local0 => val local compilerCompletions: Seq[String]
local1 => val local l$1: List[String]

Occurrences:
[3:7..3:8) <= _empty_/T.
[4:6..4:33) <= _empty_/T.enrichedCompilerCompletions().
[5:8..5:27) <= local0
[5:29..5:32) => scala/package.Seq#
[5:33..5:39) => scala/Predef.String#
[5:43..5:46) => scala/Predef.`???`().
[9:7..9:24) => _empty_/T.filterInteresting().
[11:13..11:14) <= _empty_/T.filterInteresting$default$2().(l)
[11:13..11:14) <= _empty_/T.filterInteresting().(l)
[11:16..11:20) => scala/package.List#
[11:21..11:27) => scala/Predef.String#
[12:8..12:25) <= _empty_/T.filterInteresting().
[12:26..12:34) <= _empty_/T.filterInteresting().(qualType)
[12:36..12:42) => scala/Predef.String#
[12:50..12:54) => scala/package.List#
[12:55..12:61) => scala/Predef.String#
[12:65..12:68) => scala/Predef.`???`().
```
## Expectation
Semanticdb should properly find occurences for `val l$1: List[String]#350 = compilerCompletions#52.toList#54#352` after lifting.

### PR DESCRIPTION
**Pull request by natsukagami**
## Fix #19970

- Consider the original span for synthetics for registering Ident use
- Clarify that tryFindSynthetic is for `Apply | TypeApply` tree
- Update tests

### CODE REVIEW

**Comment by rochala** [comment]
I looked at this, and I think it didn't solve the issue I've had, but the reason is completely on me. I made a mistake when I reported it. Likewise, I have updated the comment on the issue with the real reproduction. The fix should be similar, though.

**Review by rochala — CHANGES_REQUESTED** [review]
I think that the solution has to be a bit different.

Right now we're logging the occurence of local1, but we should instead go to the RHS of the ValDef from lifting. Then we should go over all calls and register their usage (they have correct spans).

Example:

```scala
object T:
  def enrichedCompilerCompletions() =
    val compilerCompletions: Seq[String] = ???

    compilerCompletions
      .toList
      .flatMap(toCompletionValues)
      .filterInteresting()


  def toCompletionValues(x: String): List[String] = ???

  extension (l: List[String])
    def filterInteresting(qualType: String = ""): List[String] = ???
``` 

results in:

```
Occurrences:
[3:7..3:8) <= _empty_/T.
[4:6..4:33) <= _empty_/T.enrichedCompilerCompletions().
[5:8..5:27) <= local0
[5:29..5:32) => scala/package.Seq#
[5:33..5:39) => scala/Predef.String#
[5:43..5:46) => scala/Predef.`???`().
[7:4..9:34) => local1
[10:7..10:24) => _empty_/T.filterInteresting().
[13:6..13:24) <= _empty_/T.toCompletionValues().
[13:25..13:26) <= _empty_/T.toCompletionValues().(x)
[13:28..13:34) => scala/Predef.String#
[13:37..13:41) => scala/package.List#
[13:42..13:48) => scala/Predef.String#
[13:52..13:55) => scala/Predef.`???`().
[15:13..15:14) <= _empty_/T.filterInteresting$default$2().(l)
[15:13..15:14) <= _empty_/T.filterInteresting().(l)
[15:16..15:20) => scala/package.List#
[15:21..15:27) => scala/Predef.String#
[16:8..16:25) <= _empty_/T.filterInteresting().
[16:26..16:34) <= _empty_/T.filterInteresting().(qualType)
[16:36..16:42) => scala/Predef.String#
[16:50..16:54) => scala/package.List#
[16:55..16:61) => scala/Predef.String#
[16:65..16:68) => scala/Predef.`???`().
```

But it should rather be:

```
Occurrences:
[3:7..3:8) <= _empty_/T.
[4:6..4:33) <= _empty_/T.enrichedCompilerCompletions().
[5:8..5:27) <= local0
[5:29..5:32) => scala/package.Seq#
[5:33..5:39) => scala/Predef.String#
[5:43..5:46) => scala/Predef.`???`().
[7:4..7:22) => local0
[8:6..8:12] => .toList
[9:6..9:xx] => .flatMap
[9:xx..9:yy] => toCompletionValue
[10:7..10:24) => _empty_/T.filterInteresting().
[13:6..13:24) <= _empty_/T.toCompletionValues().
[13:25..13:26) <= _empty_/T.toCompletionValues().(x)
[13:28..13:34) => scala/Predef.String#
[13:37..13:41) => scala/package.List#
[13:42..13:48) => scala/Predef.String#
[13:52..13:55) => scala/Predef.`???`().
[15:13..15:14) <= _empty_/T.filterInteresting$default$2().(l)
[15:13..15:14) <= _empty_/T.filterInteresting().(l)
[15:16..15:20) => scala/package.List#
[15:21..15:27) => scala/Predef.String#
[16:8..16:25) <= _empty_/T.filterInteresting().
[16:26..16:34) <= _empty_/T.filterInteresting().(qualType)
[16:36..16:42) => scala/Predef.String#
[16:50..16:54) => scala/package.List#
[16:55..16:61) => scala/Predef.String#
[16:65..16:68) => scala/Predef.`???`().
```

File: tests/semanticdb/expect/NamedApplyBlock.expect.scala [hunk 1]
```diff
@@ -4,7 +4,7 @@ object NamedApplyBlockMethods/*<-example::NamedApplyBlockMethods.*/ {
...
   def baseCase/*<-example::NamedApplyBlockMethods.baseCase().*/ = foo/*->example::NamedApplyBlockMethods.foo().*/(local/*->example::NamedApplyBlockMethods.local.*/, c/*->example::NamedApplyBlockMethods.foo().(c)*/ = 3)
-  def recursive/*<-example::NamedApplyBlockMethods.recursive().*/ = foo/*->example::NamedApplyBlockMethods.foo().*/(local/*->example::NamedApplyBlockMethods.local.*/, c/*->example::NamedApplyBlockMethods.foo().(c)*/ = foo/*->example::NamedApplyBlockMethods.foo().*/(local/*->example::NamedApplyBlockMethods.local.*/, c/*->example::NamedApplyBlockMethods.foo().(c)*/ = 3))
+  def recursive/*<-example::NamedApplyBlockMethods.recursive().*/ = foo/*->example::NamedApplyBlockMethods.foo().*//*->local1*/(local/*->example::NamedApplyBlockMethods.local.*/, c/*->example::NamedApplyBlockMethods.foo().(c)*/ = foo/*->example::NamedApplyBlockMethods.foo().*/(local/*->example::NamedApplyBlockMethods.local.*/, c/*->example::NamedApplyBlockMethods.foo().(c)*/ = 3))
```
Comment (tgodzik) [inline tests/semanticdb/expect/NamedApplyBlock.expect.scala:7]:
```suggestion
  def recursive/*<-example::NamedApplyBlockMethods.recursive().*/ = foo/*->example::NamedApplyBlockMethods.foo().*/(local/*->example::NamedApplyBlockMethods.local.*/, c/*->example::NamedApplyBlockMethods.foo().(c)*/ = foo/*->example::NamedApplyBlockMethods.foo().*/(local/*->example::NamedApplyBlockMethods.local.*/, c/*->example::NamedApplyBlockMethods.foo().(c)*/ = 3))
```
That local doesn't exist in the source code so we should not add it to occurrences

**Comment by natsukagami** [comment]
Superceded by #21856

**Pull request closed without merging**